components/infobox/wikis/smite/infobox_person_player_custom.lua

```diff
@@ -0,0 +1,96 @@
+---
+-- @Liquipedia
+-- wiki=smite
+-- page=Module:Infobox/Person/Player/Custom
+--
+-- Please see https://github.com/Liquipedia/Lua-Modules to contribute
+--
+
+local Array = require('Module:Array')
+local Class = require('Module:Class')
+local GodIcon = require('Module:GodIcon')
+local GodNames = mw.loadData('Module:GodNames')
+local Lua = require('Module:Lua')
+local Region = require('Module:Region')
+local Role = require('Module:Role')
+local String = require('Module:StringUtils')
+local TeamHistoryAuto = require('Module:TeamHistoryAuto')
+
+local Injector = Lua.import('Module:Infobox/Widget/Injector')
+local Player = Lua.import('Module:Infobox/Person')
+
+local Widgets = require('Module:Infobox/Widget/All')
+local Cell = Widgets.Cell
+local Title = Widgets.Title
+local Center = Widgets.Center
+
+local SIZE_GOD = '25x25px'
+
+---@class SmiteInfoboxPlayer: Person
+---@field role table
+---@field role2 table
+local CustomPlayer = Class.new(Player)
+local CustomInjector = Class.new(Injector)
+
+function CustomPlayer.run(frame)
+	local player = CustomPlayer(frame)
+
+	player:setWidgetInjector(CustomInjector(player))
+
+	player.args.autoTeam = true
+	player.role = Role.run({role = player.args.role})
+	player.role2 = Role.run({role = player.args.role2})
+
+	return player:createInfobox()
+end
+
+function CustomInjector:parse(id, widgets)
+	local args = self.caller.args
+	if id == 'history' then
+		local automatedHistory = TeamHistoryAuto._results{
+			convertrole = 'true',
+			iconModule = 'Module:PositionIcon/data',
+			player = self.pagename
+		}
+
+		if String.isNotEmpty(args.history) or automatedHistory then
+			return {
+				Title{name = 'History'},
+				Center{content = {args.history}},
+				Center{content = {automatedHistory}},
+			}
+		end
```

Comment:
imo for a new wiki use
```lua
	player.args.history = TeamHistoryAuto._results{
		convertrole = 'true',
		iconModule = 'Module:PositionIcon/data',
		player = self.pagename
	}
```
in run instead

Reply:
This happens after pasting it this way, but it will work if **self** is removed from **self.pagename**
![image](https://github.com/Liquipedia/Lua-Modules/assets/88981446/2bdc1936-a810-4e2f-b26d-88a543836488)